### PR TITLE
refactor(platform-browser): clean up legacy way of getting a relative path

### PR DESCRIPTION
**Pull request by crisbeto**
Currently the way we extract the pathname of a URL is by creating an anchor node, assigning the URL to its `href` and reading the `pathname`. This is inefficient and it triggers an internal security check that doesn't allow the `href` attribute to be set which ends up blocking https://github.com/angular/components/pull/28155.

These changes switch to using the browser's built-in URL parsing instead.

### CODE REVIEW

**Comment by crisbeto** [comment]
I'll TGP this tomorrow just in case.

**Review by AndrewKushnir — APPROVED** [review]
LGTM 👍

**Comment by crisbeto** [comment]
[Passing TGP](http://test/OCL:584382405:BASE:584535911:1700641186566:398ab0e3)

File: packages/platform-browser/src/browser/browser_adapter.ts [hunk 1]
```diff
@@ -89,11 +89,9 @@ function getBaseElementHref(): string|null {
...
-let urlParsingNode: HTMLAnchorElement|undefined;
-function relativePath(url: any): string {
-  urlParsingNode = urlParsingNode || document.createElement('a');
-  urlParsingNode.setAttribute('href', url);
-  const pathName = urlParsingNode.pathname;
+function relativePath(url: string): string {
+  // The base URL doesn't really matter, we just need it so relative paths have something
+  // to resolve against. In the browser `HTMLBaseElement.href` is always absolute.
+  const pathName = new URL(url, 'http://a').pathname;
   return pathName.charAt(0) === '/' ? pathName : `/${pathName}`;
```
Comment (alan-agius4) [inline packages/platform-browser/src/browser/browser_adapter.ts:96]: `new URL().pathname` always returns with a leading forward slash.

See: https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname 

```suggestion
  return new URL(url, 'http://a').pathname;
```

**Comment by crisbeto** [comment]
[Second TGP](http://test/OCL:584382405:BASE:584822513:1700730477965:4a7b5e51). This one has a couple of failing targets, but they're broken in the same way at head so I doubt that it's related to my change.

**Pull request closed without merging**